Realm/Realm/Sync/SyncTimeoutOptions.cs

```diff
@@ -0,0 +1,108 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+
+namespace Realms.Sync
+{
+    /// <summary>
+    /// Options for configuring timeouts and intervals used by the sync client.
+    /// </summary>
```

Comment:
Wouldn’t a value type be better here?

Reply:
Performance is not critical and structs are a little annoying to work with. I wanted to allow people to modify a single default, like:

```csharp
config.SyncTimeoutOptions.ConnectTimeout = TimeSpan.FromSeconds(15);
```

With a struct, you would have to replace the entire thing, making it a fair bit more verbose.